### ISSUE TITLE
dash being converted to star when typing in raw markdown in a default value

### ISSUE THREAD

**Issue by Boxersteavee** [tinacms/tinacms#4253]
### Describe the issue that you're seeing. Any Loom videos or screenshots usually help a lot!

I'm using Astro, and have a template file for each of my pages. When making a new post, using the default values I have configured, I put in a few tags using the "raw markdown" button, and typed them as such

```
- tag1
- tag2
- tag3
```
Because of the default value, that should result in the following in the markdown file
```
tags:
  - tag1
  - tag2
  - tag3
```
but instead, i got 
```
tags:
  * tag1
  * tag2
  * tag3
```
This is a problem because my configuration recognises the dashes as different list items, and I would expect the raw markdown button to copy exactly what I type into the file, but it didn't.

### Reproduction

https://github.com/Boxersteavee/website

### Steps to reproduce

1. Setup tina dev with an astro template that uses tags, like the portfolio template
2. in config.ts, add the following default value code:
```
      {
        type: "rich-text",
        name: "tags",
        label: "Tags",
        required: true,
      },
```
3. Try to modify the "tags" value in one of the using the raw markdown button (end button) and click save
4. go to the file (usually in src/content/work) and observe the tags section has been changed to have * instead of -
5. go to that page on the astro dev site, and observe the error, manually change them back to - and see it is fixed (may also have to fix a few other formatting issues, I did)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (32) x64 AMD Ryzen 9 7950X 16-Core Processor
    Memory: 15.91 GB / 31.12 GB
  Binaries:
    Node: 18.17.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - E:\Ben Project Stuff\Website\new-new-website\website\node_modules\.bin\yarn.CMD
    npm: 9.6.7 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Chromium (117.0.2045.60)
    Internet Explorer: 11.0.22621.1
  npmPackages:
    @tinacms/cli: ^1.5.30 => 1.5.30
    tinacms: ^1.5.21 => 1.5.21
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/tinacms/tinacms/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://tina.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/tinacms/tinacms/issues) that reports the same bug.

**Comment by logan-anderson** [comment]
@Boxersteavee When on the client (in the browser) markdown is stored as an "AST" in the AST we only store that an item is a list item and when it is saved to the file it looks like we use the `*` char. I don't think we store the original character that is being used only the fact that it is a list item.

This is true even when using raw mode.

The flow is something like this

Markdown raw mode -> AST (we loose info on what character was used, only the fact that is was a list item) -> sent to server -> Back to a markdown string (Converts list item to `*` )

We currently don't store what the original character was so we currently have no way to know on the backend that a `-` was used. 

I think the solution here would be to let the user chose the character that is uses for all  list items.

**Comment by Boxersteavee** [comment]
> I think the solution here would be to let the user choose the character that is used for all list items.

Yes, I think something like that would be similar. Such as allowing the user to actually write raw markdown and have that section saved exactly as it was when the user input the text.

**Comment by jamespohalloran** [comment]
For this to work, we'd likely need this to be a project-level config. I'm going to convert this to a feature request and move it over to the TinaCMS discussions
